File: go/prime_generator.go
package main

import (
	"fmt"
	"math"
	"time"
)

func isPrime(number int) bool {
	if number < 2 {
		return false
	}

	limit := int(math.Sqrt(float64(number)))

	for divisor := 2; divisor <= limit; divisor++ {
		if number%divisor == 0 {
			return false
		}
	}

	return true
}

func main() {
	startTime := time.Now()

	primes := []int{}
	candidate := 2

	for len(primes) < 10000 {
		if isPrime(candidate) {
			primes = append(primes, candidate)
		}
		candidate++
	}

	elapsed := time.Since(startTime)

	fmt.Println("Generated 10,000 primes")
	fmt.Println("Last prime:", primes[len(primes)-1])
	fmt.Printf("Elapsed time: %.4f seconds\n", elapsed.Seconds())
}

File: rust/prime_generator.rs
use std::time::Instant;

fn is_prime(number: u64) -> bool {
    if number < 2 {
        return false;
    }

    let limit = (number as f64).sqrt() as u64;

    for divisor in 2..=limit {
        if number % divisor == 0 {
            return false;
        }
    }

    true
}

fn main() {
    let start_time = Instant::now();

    let mut primes: Vec<u64> = Vec::new();
    let mut candidate: u64 = 2;

    while primes.len() < 10000 {
        if is_prime(candidate) {
            primes.push(candidate);
        }
        candidate += 1;
    }

    let elapsed = start_time.elapsed();

    println!("Generated 10,000 primes");
    println!("Last prime: {}", primes[primes.len() - 1]);
    println!("Elapsed time: {:.4} seconds", elapsed.as_secs_f64());
}
